### PR TITLE
NODE-2110: Change PoS min block time to 30s

### PR DESCRIPTION
**Pull request by Karasiq**

### CODE REVIEW

File: node/src/main/scala/com/wavesplatform/consensus/PoSCalculator.scala [hunk 1]
```diff
@@ -82,19 +82,17 @@ object NxtPoSCalculator extends PoSCalculator {
...
 
-  private val MaxSignature: Array[Byte] = Array.fill[Byte](HitSize)(-1)
-  private val MaxHit: BigDecimal        = BigDecimal(BigInt(1, MaxSignature))
-  private val C1                        = 70000
-  private val C2                        = 5e17
-  private val TMin                      = 5000
+  private[this] val MaxSignature: Array[Byte] = Array.fill[Byte](HitSize)(-1)
+  private[this] val MaxHit: BigDecimal        = BigDecimal(BigInt(1, MaxSignature))
+  private[this] val C1                        = 70000
+  private[this] val C2                        = 5e17
```
Comment (phearnot) [inline node/src/main/scala/com/wavesplatform/consensus/PoSCalculator.scala:91]: Please move all constants to `FairPoSCalculator` object.

File: node/src/main/scala/com/wavesplatform/consensus/PoSCalculator.scala [hunk 2]
```diff
@@ -120,3 +118,7 @@ object FairPoSCalculator extends PoSCalculator {
     }
   }
 }
+
+object FairPoSCalculator extends FairPoSCalculator(30000) {
+  lazy val old = new FairPoSCalculator(5000)
```
Comment (phearnot) [inline node/src/main/scala/com/wavesplatform/consensus/PoSCalculator.scala:123]: l
```suggestion
  lazy val V1 = new FairPoSCalculator(5000)
  lazy val V2 = new FairPoSCalculator(30000)
```

File: node/src/main/scala/com/wavesplatform/consensus/PoSSelector.scala [hunk 3]
```diff
@@ -20,7 +20,11 @@ import scala.concurrent.duration.FiniteDuration
 class PoSSelector(blockchain: Blockchain, blockchainSettings: BlockchainSettings, syncSettings: SynchronizationSettings) extends ScorexLogging {
   import PoSCalculator._
 
-  protected def pos(height: Int): PoSCalculator = if (fairPosActivated(height)) FairPoSCalculator else NxtPoSCalculator
+  protected def posCalculator(height: Int): PoSCalculator =
```
Comment (phearnot) [inline node/src/main/scala/com/wavesplatform/consensus/PoSSelector.scala:23]:
```suggestion
  protected def posCalculator(height: Int): PoSCalculator =
    if (vrfActivated(height)) FairPoSCalculator.V2
    else if (fairPoSActivated(height)) FairPoSCalculator.V1
    else NxtPoSCalculator
```

File: node/src/main/scala/com/wavesplatform/consensus/PoSCalculator.scala [hunk 4]
```diff
@@ -82,19 +83,22 @@ object NxtPoSCalculator extends PoSCalculator {
 
 }
 
-object FairPoSCalculator extends PoSCalculator {
+object FairPoSCalculator {
+  lazy val V1 = new FairPoSCalculator(5000)
+  lazy val V2 = new FairPoSCalculator(30000)
```
Comment (msmolyakov) [inline node/src/main/scala/com/wavesplatform/consensus/PoSCalculator.scala:88]: change to 15 seconds

**Pull request merged**